app/helpers/books_helper.rb

```diff
@@ -1,30 +1,22 @@
 module BooksHelper
 
-    def authors
-        Author.all
-    end
+  def authors
+    Author.all
+  end
 
-    def author_options
-        Author.all.collect { |c| [c.name, c.id] }.insert(0, ["--Choose author--", 0])
-    end
+  # Author.all.collect { |c| [c.name, c.id] }
+  #   => SELECT "authors".* FROM "authors"
+  # Author.all.pluck(:name, :id)
+  #   => SELECT "authors"."name", "authors"."id" FROM "authors"
+  def author_options
+    Author.all.pluck(:name, :id)
```

Comment:
[Suggestion]
If we just need to get the data from some fields (not many), we can use `pluck` instead of `map / collect`. Because 
- `pluck`: just build the SQL query to select right fields and return the array, no need to build the ActiveRecord objects.
- `map / collect`, it will select all data of all records, and build the ActiveRecord objects => it may slower and take more memory than `pluck`

https://viblo.asia/p/su-dung-hieu-qua-pluck-va-mapcollect-trong-rails-XL6lA8wNZek